jung-algorithms/src/test/java/edu/uci/ics/jung/layout/spatial/FRLayoutsTimingTest.java

```diff
@@ -70,23 +70,16 @@ public void testFRBHVisitor() {
   }
 
   private void doTest(LayoutAlgorithm<String> layoutAlgorithm) {
-    layoutModel
-        .getLayoutStateChangeSupport()
-        .addLayoutStateChangeListener(
-            new LayoutModel.LayoutStateChangeListener() {
-              long time = System.currentTimeMillis();
-
-              @Override
-              public void layoutStateChanged(LayoutModel.LayoutStateChangeEvent evt) {
-                if (evt.active == false) {
-                  long endTime = System.currentTimeMillis();
-                  log.info(
-                      "elapsed time for {} was {}",
-                      layoutAlgorithm.getClass().getName(),
-                      endTime - time);
-                }
-              }
-            });
+    long startTime = System.currentTimeMillis();
     layoutModel.accept(layoutAlgorithm);
+    layoutModel
+        .getTheFuture()
+        .thenRun(
+            () ->
+                log.info(
+                    "elapsed time for {} was {}",
+                    layoutAlgorithm.getClass().getName(),
+                    System.currentTimeMillis() - startTime))
+        .join();
   }
```

Comment:
run after the relaxer thread (and join it) instead of using the event dispatch mechanism

Reply:
why do you prefer this model?  (just curious)

Reply:
I felt that using the CompletableFuture was a real improvement over the old home-made threading code. Making the CompletableFuture accessible is useful for its 'thenRun' and 'join' methods. In this case, if I fire and receive events, I would be adding a bit more time beyond what I seek to measure.